tests/testthat/test-prepare_matches_mapper.R

```diff
@@ -7,3 +7,28 @@ test_that("`completion` column has either `low`, `medium`, or `high` values", {
   out <- prepare_matches_mapper(small_matches_mapper, ecoinvent_activities)
   expect_equal(unique(out$completion), c("low", "high", "medium"))
 })
+
+test_that("prepare_matches_mapper doesn't have any empty string", {
+  matches_mapper <- tibble(
+    ep_id = c("b", "a", "all"),
+    country = c("b", "a", "all"),
+    main_activity = c("b", "a", "all"),
+    clustered = c("b", "a", "all"),
+    activity_uuid_product_uuid = c("b", "any", NA_character_),
+    multi_match = c("b", "a", "all"),
+    completion = c("b", "a", "all"),
+    category = c("b", "a", "all")
+  )
+
+  ecoinvent_activities <- tibble(
+    activity_uuid_product_uuid = c("any", NA_character_),
+    activity_name = c(NA_character_, "c"),
+    geography = c("d", NA_character_),
+    reference_product_name = c("d", NA_character_),
+    unit = c("d", "c")
+  )
```

Comment:
The data seems to be a little larger than the minimum required. In general, tests with smaller data are easier to understand and thus maintain because the intent becomes clearer. 

My experiments suggest that we can test this with two 1-row dataframes where all values are `"a"` except for one value where `*uuid = NA_character`. This produces a failing test before we write the code, and a passing test after we write it.

We could make it easy to create such toy data with a little helper. For example:

``` r
# Helper
tibble_names <- function(x, nms) {
  m <- matrix(x, ncol = length(nms), dimnames = list(NULL, nms))
  tibble::as_tibble(m)
}
# Examples
tibble_names("a", names(cars))
#> # A tibble: 1 × 2
#>   speed dist 
#>   <chr> <chr>
#> 1 a     a
tibble_names(NA_character_, names(cars))
#> # A tibble: 1 × 2
#>   speed dist 
#>   <chr> <chr>
#> 1 <NA>  <NA>
tibble_names(c("a", NA_character_), names(cars))
#> # A tibble: 1 × 2
#>   speed dist 
#>   <chr> <chr>
#> 1 a     <NA>

# Test with `main` (to simulate the failing test before we write the code)
library(tiltIndicatorAfter)

# Arrange
mm <- tibble_names("a", names(matches_mapper))
mm$activity_uuid_product_uuid <- NA_character_
ea <- tibble_names("a", names(ecoinvent_activities))
# Act
out <- tiltIndicatorAfter:::prepare_matches_mapper(mm, ea)
# Assert
testthat::expect_equal(out$activity_uuid_product_uuid, NA_character_)
#> Error: out$activity_uuid_product_uuid (`actual`) not equal to NA_character_ (`expected`).
#> 
#> `actual`:   ""
#> `expected`: NA

# Test with this PR (to simulate the passing test after we write the code)
devtools::load_all()
#> ℹ Loading tiltIndicatorAfter

# Arrange
mm <- tibble_names("a", names(matches_mapper))
mm$activity_uuid_product_uuid <- NA_character_
ea <- tibble_names("a", names(ecoinvent_activities))
# Act
out <- tiltIndicatorAfter:::prepare_matches_mapper(mm, ea)
# Assert
testthat::expect_equal(out$activity_uuid_product_uuid, NA_character_)
```

<sup>Created on 2023-10-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
@maurolepore Yes you are right! The actual usecase can be tested with just 1-row dataframes. Thanks for giving the idea of using matrix! I have updated the test accordingly